### DISCUSSION TITLE
How to upload settings to devices with MQTT messages ?

### DISCUSSION THREAD

**Discussion by glecoz** [zwave-js/zwave-js#3584] [Q&A] [answered]
Hi,

I'm starting to migrate all my sensors to separate lightweight gateways by protocol, such as node-zwave-js, Node-Red for Enocean and RFX Com, hue2mqtt, zigbee2mqtt etc... and have most of my home logic relying on full MQTT events before summer 2022, but separately from a GUI gateway, such as Home Assistant (I will also switch to it, but prefer to decouple the devices and rely only on HA virtual devices triggered by MQTT events).

I first saw _ZWave2Mqtt_, being replaced by this project, but I can't find any documentation here about **how to upload settings to a Zwave-device with node-zwave-js from an MQTT message**. Apparently this could be done on Zwave2MQTT, as explained [here](https://github.com/OpenZWave/Zwave2Mqtt#set-values).

I especially have a few Fibaro FGMS-001-ZW5 sensors with lots of settings (see [here](https://manuals.fibaro.com/content/manuals/en/FGMS-001/FGMS-001-EN-T-v2.1.pdf) on p. 11 of user manual), that I currently update through Domoticz HTTP API.

So my questions are :

- Is the "value set" through MQTT feature still available in node-zwave-js as it was in Zwave2Mqtt ?
- Or is it a new feature here, still not implemented, and are there any plans in terms of timeline for this feature ?

I just trying to know if I can dig right now into node-zwave-js by relying on former _ZWave2Mqtt_  documentation, or if it's a best bet to still rely on Domoticz for this feature in the next months, to avoid yet another migration from ZWave2Mqtt to node-zwave-js just after a few months.

Thank you.

**Comment by AlCalzone** [comment]
@robertsLando ?

**Comment by robertsLando** [comment, marked as answer]
@glecoz Did you checked docs? https://zwave-js.github.io/zwavejs2mqtt/#/guide/mqtt?id=set-values

In this same page you will found all the MQTT related stuff

**Reply by glecoz** [reply]
Thanks @robertsLando for the info, actually I had deeply checked the docs before asking, but the first 80% of the page section from your link are about MQTT from a JS point of view, so I missed the MQTT topics for sending values to devices. I think it would be better for you to separate the JS API from the "MQTT API" and available topics in your doc, so that the latter will gain visibility and avoid useless questions (like mine, sorry ;) from users. thx again.

**Reply by robertsLando** [reply]
No worries just asking maybe you missed docs links (it already happened). BTW I dunno what you mean with JS apis, those are apis that you can call using mqtt so I had also described the signature that is needed to know the args to send in mqtt payload. It's something strictly related to mqtt but I could move it to another section, docs could be always improved :)

**Reply by glecoz** [reply]
> BTW I dunno what you mean with JS apis

Got it, I think what leads to misunderstanding is that when you land on the page, IMO, and at first view, all the documentation before [this](https://zwave-js.github.io/zwavejs2mqtt/#/guide/mqtt?id=api-call-examples) section seems to target JS developers working on the project, not on users consuming the app through MQTT, especially because the first four links on the page actually refer to JS code, and the APIs section seem to list standard JS functions from your code (especially with the example : 

![image](https://user-images.githubusercontent.com/1590118/138767130-90999815-762f-4a1c-b7f8-78adfc032e6d.png)

so I visually missed the MQTT topics in the upper parts. 

To avoid confusions, I think you could start the APIs section with a real/complete example of topic and payload, to show how you actually invoke, say the getNodes() method of your [list](https://zwave-js.github.io/zwavejs2mqtt/#/guide/mqtt?id=apis) in an MQTT message. After you answered, I googled for "node-zwave-js mqtt" and also found [this](https://github.com/zwave-js/zwavejs2mqtt/issues/1787), so yes, I think the current version of the page can be unclear at first view :)

**Reply by glecoz** [reply]
[Found](https://zwave-js.github.io/zwavejs2mqtt/#/guide/mqtt?id=api-call-examples) it, but it requires deep investigation to rebuild the puzzle, the JS code in snapshot above only belongs to the driverFunction(code) method, indenting the example under the method name may help :)

**Reply by robertsLando** [reply]
I would like to completely rewrite all the apis section making it auto-generated. BTW thanks for the tips :)
